### PR TITLE
<FileUpload/> - new component

### PR DESCRIPTION
**Pull request by bamai**
### 🔦 Summary
https://jira.wixpress.com/browse/DSM-694
...

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [ ] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [ ] Visual test

### CODE REVIEW

File: perfer.config.js [hunk 1]
```diff
@@ -140,6 +140,7 @@ const javascriptFiles = [
   ['VerticalTabs.bundle.min.js', 38],
   ['VerticalTabsItem.bundle.min.js', 38],
   ['WarningIndicator.bundle.min.js', 40],
+  ['FileUpload.bundle.min.js', 10],
```
Comment (EgoziE) [inline perfer.config.js:143]: sort and set to current value +2

File: perfer.config.js [hunk 2]
```diff
@@ -246,6 +247,7 @@ const cssFiles = [
   ['Tooltip.min.css', 2],
   ['VariableInput.min.css', 2],
   ['WarningIndicator.min.css', 2],
+  ['FileUpload.min.css', 1],
```
Comment (EgoziE) [inline perfer.config.js:250]: remove, no css here

File: src/FileUpload/docs/examples.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+export const simple = `<FileUpload><Button>upload file</Button></FileUpload>`;
+
+export const withMultipleAndAccept = `<FileUpload multiple accept=".jpeg,.gif,.png">
+<Box width="100px" height="15px" >upload files</Box>
+</FileUpload>`;
```
Comment (EgoziE) [inline src/FileUpload/docs/examples.js:5]:
```suggestion
export const simple = `
<FileUpload onChange={files => console.log(files)}>
  <Button>upload file</Button>
</FileUpload>
`;

export const withMultipleAndAccept = `
<FileUpload
  onChange={files => console.log(files)}
  multiple
  accept=".jpeg,.gif,.png"
>
  <AddItem>Add Images</AddItem>
</FileUpload>
`;
```

File: src/FileUpload/docs/index.story.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+          importExample(),
+
+          divider(),
+
+          title('Examples'),
+
+          example({
+            title: 'Simple Usage',
+            text: 'A simple example with compact preview',
```
Comment (EgoziE) [inline src/FileUpload/docs/index.story.js:66]:
```suggestion
            text: 'A single file upload button',
```

File: src/FileUpload/docs/index.story.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            source: examples.simple,
+          }),
+
+          example({
+            title: 'Upload multiple pictures',
+            text: 'FileUpload with multiple and accept',
```
Comment (EgoziE) [inline src/FileUpload/docs/index.story.js:72]:
```suggestion
            text: 'Verify only one or more image files can be uploaded',
```

File: src/FileUpload/FileUpload.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  multiple: PropTypes.bool,
+
+  /** set file types to accept */
+  accept: PropTypes.string,
+
+  /** specifies which camera to use for capture of image or video data */
+  capture: PropTypes.oneOf(['user', 'environment']),
+
+  /** input onChange callback */
+  onChange: PropTypes.func,
```
Comment (EgoziE) [inline src/FileUpload/FileUpload.js:65]: Should be required

File: src/FileUpload/FileUpload.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+FileUpload.propTypes = {
+  /** Applied as data-hook HTML attribute that can be used in the tests */
+  dataHook: PropTypes.string,
+
+  /** A css class to be applied to the component's root element */
+  className: PropTypes.string,
+
+  /** An element to be wrapped by the component */
+  children: PropTypes.element,
```
Comment (EgoziE) [inline src/FileUpload/FileUpload.js:53]: Should be required

File: src/FileUpload/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+
+export interface FileUploadProps {
+  dataHook?: string;
+  className?: string;
+  buttonText?: string;
```
Comment (EgoziE) [inline src/FileUpload/index.d.ts:6]:
```suggestion
  children: string;
  multiple?: boolean;
  accept?: string;
  capture?: 'user' | 'environment';
  onChange(fileList: FileList): void;
```

File: src/FileUpload/test/FileUpload.visual.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
```
Comment (EgoziE) [inline src/FileUpload/test/FileUpload.visual.js:1]: file can be deleted

File: stories/index.js [hunk 10]
```diff
@@ -189,3 +189,6 @@ require('./PopoverMenu/index.story'); // 7.3 PopoverMenu
 // require('../src/CustomModalLayout/docs/design.story'); // 9.3 Custom Modal (WIP)
 require('./ModalMobileLayout/index.story'); // 9.5 Mobile Modal
 require('../src/MessageBox/docs');
+
+// TODO: move to correct position
+require('../src/FileUpload/docs/index.story');
```
Comment (EgoziE) [inline stories/index.js:194]: move to correct position

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://www.wix.com/pages/wix-style-react.pr_5731/

**Pull request merged**